### PR TITLE
feat: validate identity public keys

### PR DESCRIPTION
**Pull request by jawid-h**

### CODE REVIEW

File: lib/identity/stateTransitions/identityCreateTransition/validateIdentityCreateSTDataFactory.js [hunk 1]
```diff
@@ -24,6 +25,11 @@ function validateIdentityCreateSTDataFactory(dataProvider) {
   async function validateIdentityCreateSTData(identityCreateTransition) {
     const result = new ValidationResult();
 
+    // check if public keys are valid
+    result.merge(
```
Comment (shumkov) [inline lib/identity/stateTransitions/identityCreateTransition/validateIdentityCreateSTDataFactory.js]: We can validate it without state so we could move it to the validate structure function.

**Comment by shumkov** [comment]
@jawid-h We can combine `validateDuplicatePublicKeys` and `validatePublicKeys` and name it `validatePublicKeys` because we need to call both for identity and transitions.

File: lib/identity/validation/validatePublicKeysFactory.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        if (publicKey instanceof IdentityPublicKey) {
+          return publicKey.toJSON();
+        }
+
+        return publicKey;
+      });
+
+    // validate duplication
+    result.merge(
+      validateDuplicatePublicKeys(rawPublicKeys),
```
Comment (shumkov) [inline lib/identity/validation/validatePublicKeysFactory.js]: I meant to put everything in the same function.

Comment (jawid-h) [reply lib/identity/validation/validatePublicKeysFactory.js]: @shumkov I would be a pretty fat function, isn't it better to keep them apart?

Comment (shumkov) [reply lib/identity/validation/validatePublicKeysFactory.js]: Discussed online and decided to move all validation logic (including schema) to this function.

File: lib/identity/validation/validatePublicKeysFactory.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        return publicKey;
+      });
+
+    // Validate public key structure
+    rawPublicKeys.forEach((publicKey) => {
+      result.merge(
+        validator.validate(publicKeySchema, publicKey),
+      );
+    });
+
```
Comment (shumkov) [inline lib/identity/validation/validatePublicKeysFactory.js]: If structure is invalid - return result

**Pull request merged**